doc/reference/storage_btrfs.md

```diff
@@ -1,40 +1,66 @@
-   storage pools, but if you insist then please ensure that the instance root disk's `size.state` property is set
-   to 2x the size of the root disk's size to allow all blocks in the disk image file to be rewritten without
-   reaching the qgroup quota. You may also find that using the `btrfs.mount_options=compress-force` storage pool
-   option avoids this scenario as a side effect of enabling compression is to reduce the maximum extent size such
-   that block rewrites don't cause as much storage to be double tracked. However as this is a storage pool option
-   it will affect all volumes on the pool.
-
-## Storage pool configuration
+{abbr}`Btrfs (B-tree file system)` is a single-disk file system based on the {abbr}`COW (copy-on-write)` principle.
+COW means that data is stored to a different block after it has been modified instead of overwriting the existing data, reducing the risk of data corruption.
+Unlike other file systems, Btrfs is extent-based, which means that it stores data in contiguous areas of memory.
+
+In addition to basic file system features, Btrfs offers RAID and volume management, pooling, snapshots, checksums, compression and other features.
+
+## Terminology
+
+A Btrfs file system can have *subvolumes*, which are named binary subtrees of the main tree of the file system with their own independent file and directory hierarchy.
+A *Btrfs snapshot* is a special type of subvolume that captures a specific state of another subvolume.
+Snapshots can be read-write or read-only.
+
+## `btrfs` driver in LXD
+
+The `btrfs` driver in LXD uses a subvolume per instance, image and snapshot.
+When creating a new object (for example, launching a new instance), it creates a Btrfs snapshot.
+
+Btrfs can be used as a storage backend inside a container in a nested LXD environment.
+In this case, the parent container itself must use Btrfs.
+Note, however, that the nested LXD setup does not inherit the Btrfs quotas from the parent (see {ref}`storage-btrfs-quotas` below).
+
+(storage-btrfs-quotas)=
+### Quotas
+
+Btrfs supports storage quotas via qgroups.
+Btrfs qgroups are hierarchical, but new subvolumes will not automatically be added to the qgroups of their parent subvolumes.
+This means that users can trivially escape any quotas that are set.
+Therefore, if strict quotas are needed, you should consider using a different storage driver (for example, ZFS with refquotas).
+
+When using quotas, you must take into account that Btrfs extents are immutable.
+When blocks are written, they end up in new extents.
+The old extents remain until all their data is de-referenced or rewritten.
+This means that a quota can be reached even if the total amount of space used by the current files in the subvolume is smaller than the quota.
+
+This issue is seen most often when using VMs on Btrfs, due to the random I/O nature of using raw disk image files on top of a Btrfs subvolume.
+Therefore, you should not use VMs with Btrfs storage pools.
```

Comment:
Can we emphasise this somehow?

Reply:
Perhaps we can add this whole VM part in a sub-section so we can link to it from the forums (when it inevitably comes up).

Reply:
I put it in a note now, which should emphasise it. I don't think it would work as a sub-section, since the explanation is in the Quotas section, so you'd need to link there anyway.